src/components/Login/index.jsx

```diff
@@ -1,118 +1,173 @@
 import React from "react";
+import { Row } from "@sendgrid/ui-components/grid/row";
+import { Column } from "@sendgrid/ui-components/grid/column";
+import { Button } from "@sendgrid/ui-components/button";
+import { StatefulTextInput } from "@sendgrid/ui-components/text-input";
+import { Alert } from "@sendgrid/ui-components/alert";
+import "./index.scss";
 import { Redirect } from "react-router-dom";
-import cn from "classnames";
 
 export default class Login extends React.Component {
   constructor(props) {
     super(props);
 
     this.state = {
-      isAuthenticated: false,
-      isShowingError: false,
       userName: "",
-      password: ""
+      password: "",
+      isAuthenticating: false,
+      isAuthenticated: false,
+      incorrectCredentials: false,
+      hasNetworkError: false,
     };
 
     this.handleLogin = this.handleLogin.bind(this);
   }
 
   updateField(e, field) {
     this.setState({
-      [field]: e.target.value
+      [field]: e.target.value,
     });
   }
 
   handleLogin() {
     const { userName, password } = this.state;
+    this.setState({
+      isAuthenticating: true,
+    });
 
     if (userName === "ziv" && password === "papa") {
```

Comment:
besides the only combination of correct username and password, what if user forgets to fill out one of the fields? add validation for that?